File: src/Java/Edge.java
import java.util.*;

public class Edge {
	private Node source, destination;
	private double influenceProbability;

	public Edge(Node src, Node dest, double ip) {
		this.source = src;
		this.destination = dest;
		this.influenceProbability = ip;
	}

	public Node getSource() {
		return source;
	}

	public Node getDestination() {
		return destination;
	}

	public double getIP() {
		return influenceProbability;
	}

	public boolean equals(Edge e) {
		if (this.getSource().equals(e.getSource()) && this.getDestination().equals(e.getDestination())) {
			return true;
		}
		else {
			return false;
		}
	}

	public String toString() {
		return "" + source.getID() + " " + destination.getID() + " (" + influenceProbability + ")";
	}
}

File: src/Edge.py
class Edge:
    # src: int = Source
    # dest: int = Destination
    # ip: int = probability of src influencing dest
    def __init__(self, src, dest, influence_probability, delta_probability):
        self.source = src
        self.destination = dest
        self.ip = influence_probability
        self.dp = delta_probability
    
    def equals(self, e):
        if self.source == e.source and self.destination == e.destination:
            return True
        else:
            return False
    
    def __str__(self):
        return str(self.source) + " " + str(self.destination) + " (" + str(self.ip) + ")"

    def __repr__(self):
        return self.__str__()
